mdagent/tools/base_tools/preprocess_tools/uniprot.py

```diff
@@ -0,0 +1,1250 @@
+            query: The query string to search (e.g. 'hemoglobin')
+            desired_field: The desired field to retrieve from the API
+            format_type: The format of the data to retrieve. Defaults to 'json'.
+
+        Returns:
+            The data retrieved from the API or None if no data is found.
+        """
+        with requests.Session() as session:
+            session.mount(
+                "https://",
+                HTTPAdapter(
+                    max_retries=Retry(
+                        total=5,
+                        backoff_factor=0.25,
+                        status_forcelist=[500, 502, 503, 504],
+                    )
+                ),
+            )
+            url = f"https://rest.uniprot.org/uniprotkb/search?fields={desired_field}&format={format_type}&query={query}"
+            try:
+                response = session.get(url)
+                response.raise_for_status()
+                data = response.json()
+            except requests.HTTPError:
+                print(
+                    "Requested query not found, "
+                    "please try again with a valid protein identifier."
+                )
+                return None
+            if "results" not in data or not data["results"]:
+                raise ValueError(
+                    "Requested query not found, "
+                    "please try again with a valid protein identifier."
+                )
+            return data["results"]
+
+    def _match_primary_accession(self, data: list, primary_accession: str = "") -> list:
+        """
+        Helper function to match the primary accession number with the data.
+
+        Args:
+            data: The data to search through
+            primary_accession: The primary accession number to match
+
+        Returns:
+            The relevant data entry for the primary accession number or
+                the first entry if no match is found.
+        """
+        if primary_accession:
+            matched_data = next(
+                (
+                    entry
+                    for entry in data
+                    if entry["primaryAccession"] == primary_accession
+                ),
+                None,
+            )
+            if matched_data:
+                return [matched_data]
+            print(
+                "The primary accession number provided does not "
+                "match any entry in the data, using the first entry instead."
+            )
+            return [data][0]
+        return [data][0]
+
+    def get_protein_name(
+        self,
+        query: str,
+        primary_accession: str | None = None,
+        short_names: bool = True,
+        alternative_names: bool = True,
+    ) -> list:
+        """
+        Get the protein name for a specific protein, with the option to
+        filter by primary accession number and to include alternative
+        and shortened names.
+
+        Args:
+            query: The query string to search
```

Comment:
```suggestion
            query: The query string to search (e.g. 'hemoglobin')
```

Reply:
that's actually not true, as it returns the name lol. the input query should be another identifier